.github/workflows/first-time-setup.yaml

```diff
@@ -0,0 +1,113 @@
+name: first-time-setup
+run-name: first time setup of repo
+
+on:
+  # run when called from another workflow
+  workflow_call:
+
+  # run if user manually requests it
+  workflow_dispatch:
+
+jobs:
+  first-time-setup:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Print contexts
+        uses: crazy-max/ghaction-dump-context@v1
+
+      - name: Create Pages branch
+        uses: peterjgrainger/action-create-branch@v2.4.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          branch: "gh-pages"
+
+      - name: Checkout Pages branch
+        uses: actions/checkout@v3
+        with:
+          ref: gh-pages
+
+      - name: Clear Pages branch
+        run: rm -rf * .github .gitignore
+
+      - name: Make .nojekyll file
+        run: touch .nojekyll
+
+      - name: Make placeholder homepage
+        run: echo "Your site will appear here" > index.html
+
+      - name: Commit changes to Pages branch
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          branch: gh-pages
+          commit_message: "Clear branch"
+
+      - name: Checkout main branch
+        uses: actions/checkout@v3
+
+      - name: Remove unneeded files
+        run: |
+          rm -rf \
+            README.md \
+            CHANGELOG.md \
+            testbed.md \
+            .github/ISSUE_TEMPLATE \
+            .github/workflows/versioning.yaml \
+            .github/pull_request_template.md \
```

Comment:
AHH! The magic!

Reply:
Indeed. And at one point, I had this "first time setup" replace the PR template with a user-specific one. And in the GitHub UI, it shows the template of the _destination_ repo, not the source.